django/db/models/query.py

```diff
@@ -1168,10 +1168,24 @@ def only(self, *fields):
             # Can only pass None to defer(), not only(), as the rest option.
             # That won't stop people trying to do this, so let's be explicit.
             raise TypeError("Cannot pass None as an argument to only().")
+
+        reverse_fields = set(
```

Comment:
This can use set comprehension

 ```suggestion
         reverse_fields = {
```